From 9a1ea149d3451d4edf32c5f0f0d1b4fecfbff4a2 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 6 Aug 2019 22:11:57 +0200
Subject: [PATCH] config.mk: add WITH_TLS_STATIC_LIB_DEPS

Add WITH_TLS_STATIC_LIB_DEPS to allow the user to give openssl static
libraries dependencies such as -lz or -latomic

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://github.com/eclipse/mosquitto/pull/1371]
---
 config.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/config.mk b/config.mk
index 6c23bef..c9e2daf 100644
--- a/config.mk
+++ b/config.mk
@@ -23,6 +23,10 @@
 # password authentication at all.
 WITH_TLS:=yes
 
+# Use this variable to give openssl static libraries dependencies (such as -lz
+# or -latomic) that can be retrieved for example by pkg-config
+WITH_TLS_STATIC_LIB_DEPS:=
+
 # Comment out to disable TLS/PSK support in the broker and client. Requires
 # WITH_TLS=yes.
 # This must be disabled if using openssl < 1.0.
@@ -202,7 +206,7 @@ ifeq ($(WITH_TLS),yes)
 	LIB_CPPFLAGS:=$(LIB_CPPFLAGS) -DWITH_TLS
 	PASSWD_LDADD:=$(PASSWD_LDADD) -lcrypto
 	CLIENT_CPPFLAGS:=$(CLIENT_CPPFLAGS) -DWITH_TLS
-	STATIC_LIB_DEPS:=$(STATIC_LIB_DEPS) -lssl -lcrypto
+	STATIC_LIB_DEPS:=$(STATIC_LIB_DEPS) -lssl -lcrypto $(WITH_TLS_STATIC_LIB_DEPS)
 
 	ifeq ($(WITH_TLS_PSK),yes)
 		BROKER_CPPFLAGS:=$(BROKER_CPPFLAGS) -DWITH_TLS_PSK
-- 
2.20.1

